### PR TITLE
Add autoYaST for caasp

### PR DESCRIPTION
**Pull request by dzedro**
[test](http://10.100.12.155/tests/4861)

### CODE REVIEW

**Comment by kravciak** [comment]
Trevis probably does not like that you redefined load_autoyast_tests from main_common.pm
Do you want separate sub for 1 test module?
@okurz, I see https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/2348 is open, any relation?

File: products/casp/main.pm [hunk 1]
```diff
@@ -103,19 +108,27 @@ sub load_feature_tests {
...
-if (check_var('FLAVOR', 'DVD')) {
-    if (get_var('EXTRA', '') =~ /RCSHELL/) {
-        load_rcshell_tests;
-        return 1;
+    if (check_var('FLAVOR', 'DVD')) {
+        if (get_var('EXTRA', '') =~ /RCSHELL/) {
+            load_rcshell_tests;
+            return 1;
+        }
+        load_inst_tests;
```
Comment (kravciak) [inline products/casp/main.pm]: Please keep main.pm simple by reusing existing structure. All you need to change is line 125 to:
if (get_var('AUTOYAST')) {
  loadtest 'autoyast/installation';
} else {
  load_inst_tests;
}

**Comment by kravciak** [comment]
Please provide soft_fail with bug number when you use AUTOYAST_EXPECT_ERRORS=1
In your test run everything is green but you have errors during installation. (as discussed)

File: products/casp/main.pm [hunk 2]
```diff
@@ -103,19 +108,27 @@ sub load_feature_tests {
 }
 
 # ==== Installation workflow ====
-load_boot_tests;
+if (get_var('AUTOYAST')) {
+    load_boot_tests;
+    load_autoyast_tests;
+    loadtest 'casp/first_boot';
+    load_feature_tests;
```
Comment (kravciak) [inline products/casp/main.pm]: If you want to run feature tests after autoyast installation set test variable EXTRA=FEATURES

File: tests/autoyast/installation.pm [hunk 3]
```diff
@@ -111,6 +111,8 @@ sub run {
             last if match_has_tag("bios-boot") || match_has_tag("reboot-after-installation");
 
             if (match_has_tag('autoyast-error')) {
+                my $reason = get_var('AUTOYAST_EXPECT_ERRORS_REASON');
+                record_soft_failure "AUTOYAST_EXPECT_ERRORS $reason";
```
Comment (okurz) [inline tests/autoyast/installation.pm]: I think `record_soft_failure 'AUTOYAST_EXPECT_ERRORS ' . get_var('AUTOYAST_EXPECT_ERRORS_REASON');` should be enough

**Comment by kravciak** [comment]
Please set up AUTOYAST_EXPECT_ERRORS_REASON with bug reference afterwards

**Pull request merged**